crates/composer/src/jukebox.rs

```diff
@@ -0,0 +1,60 @@
+use eyre::{Context, Result};
+use rodio::{
+    source::{Buffered, SamplesConverter},
+    Decoder, OutputStreamHandle, Source,
+};
+use std::{fs::File, io::BufReader, path::Path};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) enum Sample {
+    Click,
+}
+
+impl Sample {
+    const ALL: &[Sample] = &[Self::Click];
+
+    fn filename(&self) -> &'static Path {
+        Path::new(match self {
+            Sample::Click => "click.wav",
+        })
+    }
+}
+
+type Buffer = Buffered<SamplesConverter<Decoder<BufReader<File>>, f32>>;
```

Comment:
This is a bit hairy, but I *think* it does what we want: read the samples from disk on the first play, and any subsequent play reads them from the buffer. (.clone()ing this should be cheap)

Reply:
Agree this is fine.